script.js

```diff
@@ -1,4 +1,368 @@
+  }
+  // If in LOWEST_COMBINED_NUMBER mode, display the player with the larger score first
+  if (mode == LOWEST_COMBINED_NUMBER) {
+    // If the smaller score is in index 0, loop thru the array and add-on to leaderBoardOutput
+    if (playerProfiles[0].score < playerProfiles[1].score) {
+      for (var i = 0; i < playerProfiles.length; i += 1) {
+        // In each iteration of the loop, add on to the preamble by displaying the player name and scores
+        leaderBoardOutput +=
+          'Player ' +
+          playerProfiles[i].id +
+          ': ' +
+          playerProfiles[i].score +
+          '<br>';
+      }
+    }
+    // If the smaller score is in index 1, loop from the back of the array and add-on to leaderBoardOutput
+    else if (playerProfiles[1].score < playerProfiles[0].score) {
+      for (var i = playerProfiles.length - 1; i > -1; i -= 1) {
+        // In each iteration of the loop, add on to the preamble by displaying the player name and scores
+        leaderBoardOutput +=
+          'Player ' +
+          playerProfiles[i].id +
+          ': ' +
+          playerProfiles[i].score +
+          '<br>';
+      }
+    }
+    return leaderBoardOutput;
+  }
+};
+
+/**
+ * Reset the game by re-initialising relevant variables
+ */
+var resetGame = function () {
+  currPlayer = 1;
+  gameMode = GAME_MODE_DICE_ROLL;
+};
+
+/*====================================
+==============MAIN====================
+======================================*/
+/**
+ * Play Beat That as per SWE101 game rules
+ * https://swe101.rocketacademy.co/projects/project-2-dice
+ * @param {string} input
+ */
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // ------------MORE COMFORTABLE------------------
+  if (!mode) {
+    // Validet that input is one of the modes
+    if (input != REGULAR && input != LOWEST_COMBINED_NUMBER) {
+      return (
+        'Please choose one of the following modes:<br> 1. ' +
+        REGULAR +
+        '<br> 2. ' +
+        LOWEST_COMBINED_NUMBER
+      );
+    }
+    // if the input is valid, re-assign mode with the input value
+    mode = input;
+    return (
+      'You have selected ' +
+      mode +
+      ' mode. Player ' +
+      currPlayer +
+      ', click submit to roll your dice'
+    );
+  }
+
+  if (gameMode === GAME_MODE_DICE_ROLL) {
+    // Roll 2 dice and show the player the values
+    // Get dice rolls for curr player and populate the curr player's dice array
+    var newDiceRolls = getDiceRolls();
+    // Switch mode to choose dice order
+    gameMode = GAME_MODE_CHOOSE_DICE_ORDER_AUTOMATICALLY;
+    // Return the dice roll values to that player
+    return (
+      'Welcome Player ' +
+      currPlayer +
+      '. <br>You rolled Dice 1: ' +
+      newDiceRolls[0] +
+      ' and Dice 2: ' +
+      newDiceRolls[1] +
+      '<br>    Click submit to see your number'
+    );
+  }
+
+  //Automatically generate the largest possible number based on a player's dice roll
```

Comment:
Space after `//` for consistency

Reply:
@jeremylim91 are you using prettier when coding? thought prettier would have fixed this automatically